app/scripts/controllers/push-platform-notifications/services/services.ts

```diff
@@ -195,18 +195,21 @@ export async function listenToPushNotifications(
   const unsubscribePushNotifications = onBackgroundMessage(
     messaging,
     async (payload: MessagePayload): Promise<void> => {
-      const typedPayload = payload;
-
-      // if the payload does not contain data, do nothing
       try {
-        const notificationData: NotificationUnion = typedPayload?.data?.data
-          ? JSON.parse(typedPayload?.data?.data)
+        const data = payload?.data?.data
+          ? JSON.parse(payload?.data?.data)
           : undefined;
 
-        if (!notificationData) {
+        // if the payload does not contain data, do nothing
+        if (!data) {
           return;
         }
 
+        const notificationData = {
+          ...data,
+          type: data?.type ?? data?.data?.kind,
+        } as NotificationUnion;
```

Comment:
God more bad types we need to resolve.

A raw notification DOES NOT have a `type` property, this is something we inject before processing. This adds back this type property if it is missing.

Reply:
Most of our types are sound. These are edge-cases and artefacts from the portfolio integration.